packages/jellyfish-api-core/__tests__/category/masternode/setGov.locks.token.test.ts

```diff
@@ -0,0 +1,585 @@
+    await testing.generate(1)
+  }
+
+  async function futureSwapSetup (): Promise<void> {
+    await depositToVaultAndTakeLoan()
+
+    await testing.rpc.account.sendTokensToAddress({}, { [collateralAddress]: ['1@TSLA'] })
+    await testing.generate(1)
+
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/params/dfip2203/active': 'false' } })
+    await testing.generate(1)
+
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/params/dfip2203/reward_pct': '0.05', 'v0/params/dfip2203/block_period': '25' } })
+    await testing.generate(1)
+
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/params/dfip2203/active': 'true' } })
+    await testing.generate(1)
+  }
+
+  async function loan1Setup (): Promise<void> {
+    await depositToVaultAndTakeLoan()
+
+    await testing.rpc.loan.setCollateralToken({
+      token: 'TSLA',
+      factor: new BigNumber(1),
+      fixedIntervalPriceId: 'TSLA/USD'
+    })
+    await testing.generate(1)
+  }
+
+  async function loan2Setup (): Promise<void> {
+    await loan1Setup()
+
+    await testing.rpc.vault.depositToVault({
+      vaultId, from: collateralAddress, amount: '1@TSLA'
+    })
+    await testing.generate(1)
+  }
+
+  async function loan3Setup (): Promise<void> {
+    await depositToVaultAndTakeLoan()
+
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/token/${tslaId}/payback_dfi`]: 'true', [`v0/token/${tslaId}/payback_dfi_fee_pct`]: '0.02' } })
+    await testing.generate(1)
+  }
+
+  it('should lock and unlock loan token', async () => {
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`]).toBeUndefined()
+    }
+
+    // Lock loan token
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`]).toStrictEqual('true')
+    }
+
+    // Unlock loan token
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`]).toStrictEqual('false')
+    }
+
+    // Lock and unlock loan tokens in the same block
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'true' ||
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'false'
```

Comment:
Looks like `attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`]` should be always false, but sometimes it becomes true, is it acceptable ?

Reply:
depends on the order of txns in the block.
anw setgov requires foundation auth and is highly unlikely to do ON/OFF in the same block.